File: php/features.php
<?php
declare(strict_types=1);

// Navalnyarchive SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class NavalnyarchiveFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new NavalnyarchiveBaseFeature();
            case "test":
                return new NavalnyarchiveTestFeature();
            default:
                return new NavalnyarchiveBaseFeature();
        }
    }
}

File: py/navalnyarchive_sdk/features.py
# Navalnyarchive SDK feature factory

from navalnyarchive_sdk.feature.base_feature import NavalnyarchiveBaseFeature
from navalnyarchive_sdk.feature.test_feature import NavalnyarchiveTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NavalnyarchiveBaseFeature(),
        "test": lambda: NavalnyarchiveTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
